### ISSUE TITLE
Clavister NetWall

### ISSUE THREAD

**Issue by DimitriPapadopoulos** [dlenski/what-vpn#23]
Have you heard of [Clavister NetWall](https://www.clavister.com/products/ngfw/)? Appears to be somehow compatible with AnyConnect SSL VPN.

See [Clavister (Classic) SSL VPN vs OneConnect (OpenConnect based) SSL VPN](https://kb.clavister.com/332433702/clavister-classic-ssl-vpn-vs-oneconnect-openconnect-based-ssl-vpn).

**Comment by dlenski** [comment]
Hmmm… no I haven't. Their OneConnect sounds like it might simply be a wrapper around ocserv. I wonder how well they're complying with its license. 🧐

**Comment by dlenski** [comment]
Do you know of publicly-accessible Clavister servers? What does `what-vpn` say when pointed at them?

**Comment by DimitriPapadopoulos** [comment]
No, I haven't found any. Not sure how to find any with help from Google or other search engines.

* [OneConnect](https://www.clavister.com/products/oneconnect/) are the VPN clients.
* [NetWall NGFW](https://www.clavister.com/products/ngfw/) are the firewalls with “[Reliable Secure VPN](https://www.clavister.com/use-cases/reliable-secure-vpn/)”.

There is no evidence they use OpenConnect code, client side or server side. They do share the same AnyConnect protocol (or OpenConnect protocol as they call it) and OpenConnect is clearly on their radar:
* [Configure the OpenConnect-GUI client towards Clavister NetWall](https://kb.clavister.com/329092217/configure-the-openconnect-gui-client-towards-clavister-netwall)
* [Configure the Android OpenConnect client towards Clavister NetWall](https://kb.clavister.com/329095020/configure-the-android-openconnect-client-towards-clavister-netwall)
* [Configure Linux OpenConnect towards Clavister NetWall](https://kb.clavister.com/329092224/configure-linux-openconnect-towards-clavister-netwall)
* [Clavister (Classic) SSL VPN vs OneConnect (OpenConnect based) SSL VPN](https://kb.clavister.com/332433702/clavister-classic-ssl-vpn-vs-oneconnect-openconnect-based-ssl-vpn)

It would be nice to be able to use their iOS, Android and Windows OneConnect clients to connect to ocserv servers, since we lack well-maintained clients for these platforms. Unfortunately, it looks like there are some inconsistencies between NetWall and ocserv (which supports the idea the code base is different):
https://gitlab.com/openconnect/ocserv/-/issues/485

It would also be nice to test whether OpenConnect can indeed connect to Clavister NetWall appliances.

**Comment by DimitriPapadopoulos** [comment]
The [Clavister OneConnect Android client](https://play.google.com/store/apps/details?id=com.clavister.oneconnect) uses [wolfSSL](https://www.wolfssl.com/) and [Apache HttpComponents](https://hc.apache.org/) as far as I can see by looking into the [APK file](https://apkpure.com/fr/clavister-oneconnect/com.clavister.oneconnect), so it does seem they have rewritten the client at least.
```
$ unzip -q Clavister\ OneConnect_3.5_Apkpure.xapk
$ 
$ unzip -t config.arm64_v8a.apk | grep -i wolf
    testing: lib/arm64-v8a/libwolfssl.so   OK
    testing: lib/arm64-v8a/libwolfsslwrapper.so   OK
$ 
$ unzip -t com.clavister.oneconnect.apk | grep -i apache/hc
    testing: org/apache/hc/client5/version.properties   OK
    testing: org/apache/hc/core5/version.properties   OK
$
